### PR TITLE
Feature canbuspacket is ready to be merged with release.

### PR DESCRIPTION
**Pull request by ThePatman3**
The Feature-canbuspacket branch was recreated to implement canbus functionality in the ```module_c``` class. The old packet is still being send in this branch, the new packet will be implemented in another branch after we have created interfaces for the lidar and sonar.

### CODE REVIEW

File: code/headers/module.hpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    /// This class uses a lidar to measure the distance between the lidar and
+    /// the closest object. When a distance request has been received this class
+    /// sends the smallest detected distance over the can-bus.
+    class module_c : public base_module_c {
+    private:
+        lidar_c& lidar;
+
+        uint16_t smallest_value;
+        uint16_t current_value;
+        bool smallest_value_set_first_time = false;
```
Comment (Gian1080) [inline code/headers/module.hpp:21]: Tried to understand this bool, not sure what it does, sure it works, care to explain?

**Review by Gian1080 — COMMENTED** [review]
Not 100% sure, but I think that doxygen requires 3 forward slashes for the comments at the variables.

File: code/headers/module.hpp (hunk 1 above)
Comment (ThePatman3) [reply code/headers/module.hpp:21]: The module sends the smallest distance value on the canbus. To do this, all measured distances are compared with the last smallest distance encountered. This however raises the problem of what the initial smallest value is, before any measurements are made. If the largest possible value is used, what happens when only failed measurements (with a value of 0) are received? Then this largest value will be send on the canbus, while that is not a valid measurement. To make sure only a valid measurement is used in the comparison, a bool, ```smallest_value_set_first_time ```, is used to keep track of whether a valid measurement has been received that can be used in the comparison.

Comment (Gian1080) [reply code/headers/module.hpp:21]: Makes sense, thank you for the explanation!

**Review by Gian1080 — COMMENTED** [review]
Good code, well explained, stay frosty!

**Review by Gian1080 — APPROVED** [review]
Correction, clicked the wrong option.

**Pull request merged**